src/frontends/jax/src/op/reduce_window_sum.cpp

```diff
@@ -0,0 +1,99 @@
+    auto base_dilation = context.const_named_param<std::vector<int64_t>>("base_dilation");
+    auto window_dilation = context.const_named_param<std::vector<int64_t>>("window_dilation");
+    size_t total_dim = window_dimensions.size();
+
+    JAX_OP_CONVERSION_CHECK(window_strides.size() == total_dim,
+                            "Internal error: window_strides must have the same size as window_dimensions, but got " +
+                                std::to_string(window_strides.size()) + " and " + std::to_string(total_dim));
+    JAX_OP_CONVERSION_CHECK(padding.size() == total_dim,
+                            "Internal error: padding must have the same size as window_dimensions, but got " +
+                                std::to_string(padding.size()) + " and " + std::to_string(total_dim));
+    JAX_OP_CONVERSION_CHECK(base_dilation.size() == total_dim,
+                            "Internal error: base_dilation must have the same size as window_dimensions, but got " +
+                                std::to_string(base_dilation.size()) + " and " + std::to_string(total_dim));
+    JAX_OP_CONVERSION_CHECK(window_dilation.size() == total_dim,
+                            "Internal error: window_dilation must have the same size as window_dimensions, but got " +
+                                std::to_string(window_dilation.size()) + " and " + std::to_string(total_dim));
+
+    Strides strides(total_dim - 2);
+    Shape pads_begin(total_dim - 2);
+    Shape pads_end(total_dim - 2);
+    Shape kernel(total_dim - 2);
+    for (size_t ind = 0; ind < total_dim; ++ind) {
+        if (ind != 0 && ind != total_dim - 1) {
+            kernel[ind - 1] = static_cast<size_t>(window_dimensions[ind]);
+            pads_begin[ind - 1] = padding[ind][0];
+            pads_end[ind - 1] = padding[ind][1];
+            strides[ind - 1] = static_cast<size_t>(window_strides[ind]);
+        } else {
+            // only support NHWC format input now.
+            JAX_OP_CONVERSION_CHECK(window_dimensions[ind] == 1, "Internal error: unsupported layout.");
+            JAX_OP_CONVERSION_CHECK(window_strides[ind] == 1, "Internal error: unsupported layout.");
+            JAX_OP_CONVERSION_CHECK(padding[ind][0] == 0 && padding[ind][1] == 0,
+                                    "Internal error: unsupported layout.");
+            JAX_OP_CONVERSION_CHECK(base_dilation[ind] == 1, "Internal error: unsupported layout.");
+        }
+        JAX_OP_CONVERSION_CHECK(window_dilation[ind] == 1, "Internal error: only window_dilation 1 is supported.");
+    }
+
+    std::vector<int64_t> in_transpose_vector(total_dim);
+    std::vector<int64_t> out_transpose_vector(total_dim);
+    in_transpose_vector[0] = 0;
+    in_transpose_vector[1] = total_dim - 1;
+    out_transpose_vector[0] = 0;
+    out_transpose_vector[total_dim - 1] = 1;
+    for (size_t i = 2; i < total_dim; i++) {
+        in_transpose_vector[i] = i - 1;
+        out_transpose_vector[i - 1] = i;
+    }
```

Comment:
This can be a helper function in `utils.hpp`. It looks like you will need it more then once